src/utils/codemods/findAvailableCodemodsInNodeModules.js

```diff
@@ -2,32 +2,81 @@ const log = require('@dhis2/cli-helpers-engine').reporter
 const fs = require('fs-extra')
 const path = require('path')
 
-const isNotDotFolder = value => value !== '.' && value !== '..'
+const isDotFolder = name => name[0] === '.'
+
+const isNotDotFolder = name => !isDotFolder(name)
 
 const isDotJSFile = name => new RegExp('[.]js$').test(name)
 
-const extractCodemods = (paths, moduleName) => {
-    const { DHIS2_NODE_MODULES } = paths
-    const codemodsPath = path.join(DHIS2_NODE_MODULES, moduleName, 'codemods')
+const isFolder = path => {
+    if (!fs.existsSync(path)) return false
+    return fs.lstatSync(path).isDirectory()
+}
 
-    if (!fs.pathExistsSync(codemodsPath)) {
-        log.debug(
-            `Cound not find a codemods folder in node module '${moduleName}' directory`
-        )
+const hasCodemods = codemodsPath =>
+    !!fs.readdirSync(codemodsPath).filter(curFile => {
+        const curPath = path.join(codemodsPath, curFile)
 
-        return []
-    }
+        if (isFolder(curPath)) return false
+        if (!curFile.match(/\.(j|t)sx?$/)) return false
+
+        return true
+    }).length
+
+/**
+ * Tries to find the following dirs that contain .js files:
+ *   - ${NODE_MODULES}/{{folder}}/codemods/
+ *   - ${NODE_MODULES}/{{namespace}}/{{folder}}/codemods/
+ *
+ * @param {Object} paths
+ * @returns {String[]}
+ */
+const findCodemodsFolders = paths => {
+    return fs
+        .readdirSync(paths.NODE_MODULES)
+        .filter(name => {
+            const curPath = path.join(paths.NODE_MODULES, name)
+
+            if (isDotFolder(name)) return false
+            if (!isFolder(curPath)) return false
+
+            return true
+        })
+        .reduce((acc, cur) => {
+            const curPath = path.join(paths.NODE_MODULES, cur)
+
+            // Node module is not namespace and contains codemods
+            if (
+                isFolder(path.join(curPath, 'codemods')) &&
+                hasCodemods(path.join(curPath, 'codemods'))
+            ) {
+                return [...acc, cur]
+            }
+
+            // Node module is a namespace containing node modules that contain codemods
+            const curPathFolders = fs
+                .readdirSync(curPath)
+                .filter(
+                    name =>
+                        isNotDotFolder(name) &&
+                        isFolder(path.join(curPath, name, 'codemods')) &&
+                        hasCodemods(path.join(curPath, name, 'codemods'))
+                )
+                .map(name => `${cur}/${name}`)
```

Comment:
I *think* this is actually incorrect logic... namespaces in npm must always start with @, but this would also traverse a second level of the node_modules directory tree for modules that just don't have their own codemods dir - the first condition would fail for i.e. `lodash` because there's no `codemods` directory there, but the algorithm would continue to traverse all sub-directories in the `lodash` package looking for one that contains `codemods`.  This could result in false-positives (though unlikely) and would also be quite expensive.